### PR TITLE
From nsidc cumulus 3220: Create a new send-pan task

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-3220: Create a new send-pan task for use in SIPS workflows](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3220)

Original user PR: https://github.com/nasa/cumulus/pull/3320
## Changes

* Created a new send-pan task

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

File: packages/api/lib/pdrHelpers.js
Comment (jennyhliu) [inline packages/api/lib/pdrHelpers.js:1]: Add back the methods which are removed from v16:
https://github.com/nasa/cumulus/blob/release-15.0.x/packages/api/models/pdrs.js

File: example/spec/parallel/ingest/ingestFromPdrSpec.js [hunk 1]
```diff
@@ -516,6 +517,32 @@ describe('Ingesting from PDR', () => {
...
+          const panName = lambdaOutput.payload.pdr.name.replace(/\.pdr/gi, '.pan');
+          const panKey = path.join(addedCollections[0].meta.panPath, panName);
+          const expectedPanUri = buildS3Uri(config.bucket, panKey);
+          expect(lambdaOutput.payload.pan.uri).toEqual(expectedPanUri);
+          const panExists = await s3ObjectExists({
+            Bucket: config.bucket,
+            Key: panKey,
+          });
+          expect(panExists).toEqual(true);
+          deleteS3Object(config.bucket, panKey);
```
Comment (botanical) [inline example/spec/parallel/ingest/ingestFromPdrSpec.js:543]:
```suggestion
          await deleteS3Object(config.bucket, panKey);
```

File: example/spec/parallel/ingest/ingestPdrWithNodeNameSpec.js [hunk 2]
```diff
@@ -512,7 +508,7 @@ describe('Ingesting from PDR', () => {
 
             // the output of the CheckStatus is used to determine the task of choice
             const checkStatusTaskName = 'CheckStatus';
-            const successStepName = 'WorkflowSucceeded';
+            const successStepName = 'SendPAN';
```
Comment (botanical) [inline example/spec/parallel/ingest/ingestPdrWithNodeNameSpec.js:511]: Similar to ingestFromPdrSpec, should we have some assertions that the SendPAN task was invoked and successful?

File: packages/ingest/src/S3ProviderClient.ts [hunk 3]
```diff
@@ -79,8 +79,8 @@ class S3ProviderClient implements ProviderClient {
   /**
    * Download the remote file to a given s3 location
    *
-   * @param {Object} params - the full path to the remote file to be fetched
-   * @param {string} params.sourceKey - the full path to the remote file to be fetched
+   * @param {object} params - the full path to the remote file to be fetched
+   * @param {string} params.fileRemotePath - the full path to the remote file to be fetched
```
Comment (botanical) [inline packages/ingest/src/S3ProviderClient.ts:83]: ⭐

File: packages/ingest/src/S3ProviderClient.ts [hunk 4]
```diff
@@ -145,6 +145,34 @@ class S3ProviderClient implements ProviderClient {
     }
   }
 
+  /**
+   * Upload a file
+   *
+   * @param {object} params
+   * @param {string} params.localPath - the full local file path
+   * @param {string} params.uploadPath - the full remote file path for uploading file to
+   * @param {string} params.remoteAltBucket - alternate per-file bucket override to this.bucket
```
Comment (botanical) [inline packages/ingest/src/S3ProviderClient.ts:154]:
```suggestion
   * @param {string} [params.remoteAltBucket] - alternate per-file bucket override to this.bucket
```
https://jsdoc.app/tags-param.html

File: packages/ingest/src/HttpProviderClient.js [hunk 5]
```diff
@@ -206,7 +207,7 @@ class HttpProviderClient {
   /**
    * Download a remote file to disk
    *
-   * @param {Object} params
+   * @param {object} params
```
Comment (botanical) [inline packages/ingest/src/HttpProviderClient.js:210]:
```suggestion
   * @param {Object} params
```

File: packages/ingest/src/HttpProviderClient.js [hunk 6]
```diff
@@ -244,7 +245,7 @@ class HttpProviderClient {
   /**
    * Download the remote file to a given s3 location
    *
-   * @param {Object} params
+   * @param {object} params
```
Comment (botanical) [inline packages/ingest/src/HttpProviderClient.js:248]:
```suggestion
   * @param {Object} params
```

File: packages/ingest/src/HttpProviderClient.js [hunk 7]
```diff
@@ -291,6 +292,39 @@ class HttpProviderClient {
     return { s3uri, etag };
   }
 
+  /**
+   * Upload a file
+   *
+   * @param {object} params
```
Comment (botanical) [inline packages/ingest/src/HttpProviderClient.js:298]:
```suggestion
   * @param {Object} params
```

File: packages/ingest/src/S3ProviderClient.ts [hunk 8]
```diff
@@ -15,7 +15,7 @@ class S3ProviderClient implements ProviderClient {
   /**
    * Download a remote file to disk
    *
-   * @param {Object} params
+   * @param {object} params
```
Comment (botanical) [inline packages/ingest/src/S3ProviderClient.ts:18]:
```suggestion
   * @param {Object} params
```

File: packages/ingest/src/S3ProviderClient.ts [hunk 9]
```diff
@@ -79,8 +79,8 @@ class S3ProviderClient implements ProviderClient {
   /**
    * Download the remote file to a given s3 location
    *
-   * @param {Object} params - the full path to the remote file to be fetched
-   * @param {string} params.sourceKey - the full path to the remote file to be fetched
+   * @param {object} params - the full path to the remote file to be fetched
```
Comment (botanical) [inline packages/ingest/src/S3ProviderClient.ts:82]:
```suggestion
   * @param {Object} params - the full path to the remote file to be fetched
```

File: packages/ingest/src/S3ProviderClient.ts [hunk 10]
```diff
@@ -145,6 +145,34 @@ class S3ProviderClient implements ProviderClient {
     }
   }
 
+  /**
+   * Upload a file
+   *
+   * @param {object} params
```
Comment (botanical) [inline packages/ingest/src/S3ProviderClient.ts:151]:
```suggestion
   * @param {Object} params
```

File: tasks/send-pan/src/index.ts [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    default:
+      throw new Error(`Protocol ${providerConfig.protocol} is not supported.`);
+  }
+};
+
+/**
+ * Send PAN and return the uri of the uploaded pan
+ *
+ * @param {object} event - input from the message adapter
+ * @returns {object} the uri of the pan
```
Comment (botanical) [inline tasks/send-pan/src/index.ts:38]:
```suggestion
 * @param {Object} event - input from the message adapter
 * @returns {Object} the URI of the pan
```

File: tasks/send-pan/src/index.ts [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+    pan: { uri },
+  };
+}
+
+/**
+ * Lambda handler
+ *
+ * @param {object} event      - a Cumulus Message
+ * @param {object} context    - an AWS Lambda context
+ * @returns {Promise<object>} - Returns output from task.
```
Comment (botanical) [inline tasks/send-pan/src/index.ts:69]:
```suggestion
 * @returns {Promise<CumulusMessage | CumulusRemoteMessage>} - Returns output from task.
```
I believe it should be returning a CumulusMessage?

**Review by botanical — APPROVED** [review]
Thanks for ushering in these changes @jennyhliu 🙇

**Pull request merged**